DEMOQA/cypress/tests/login.js

```diff
@@ -0,0 +1,204 @@
+            cy.get("#userNumber").should('have.css', 'border-color', 'rgb(40, 167, 69)')
+            cy.get('#dateOfBirthInput').should('have.css', 'border-color', 'rgb(40, 167, 69)')
+            cy.get('label[for="hobbies-checkbox-1"]').should('have.css', 'border-color', 'rgb(40, 167, 69)')
+            cy.get('textarea[placeholder="Current Address"]').should('have.css', 'border-color', 'rgb(40, 167, 69)')
+        })
+    })
+
+    it('submit successfully with a needed input', () => {
+        cy.fixture('testData').then((data) => {
+            cy.get("#firstName").type(data.firstName)
+            cy.get("#lastName").type(data.lastName)
+            cy.get("label[for='gender-radio-1']").click();
+            cy.get("#userNumber").type(data.userNumber)
+
+            cy.get("#submit").click({ force: true })
+
+            cy.get(".modal-content").should("be.visible");
+            cy.get("#firstName").should('have.css', 'border-color', 'rgb(40, 167, 69)')
+            cy.get("#lastName").should('have.css', 'border-color', 'rgb(40, 167, 69)')
+            cy.get("label[for='gender-radio-1']").should('have.css', 'border-color', 'rgb(40, 167, 69)')
+            cy.get("#userNumber").should('have.css', 'border-color', 'rgb(40, 167, 69)')
+        })
+    })
+
+    it('Submit successfully with mobile start with code', () => {
+
+        cy.fixture('testData').then((data) => {
+            cy.get("#firstName").type(data.firstName)
+            cy.get("#lastName").type(data.lastName)
+            cy.get("label[for='gender-radio-1']").click();
+            cy.get("#userNumber").type("+8493300884")
+
+            cy.get("#submit").click({ force: true })
+
+            cy.get("#userNumber").should('have.css', 'border-color', 'rgb(220, 53, 69)');
+
+        })
+    })
+
+    it('cannot submit with all empty', () => {
+
+        cy.get("#submit").click({ force: true })
+
+        cy.get("#firstName").should('have.css', 'border-color', 'rgb(220, 53, 69)')
+        cy.get("#lastName").should('have.css', 'border-color', 'rgb(220, 53, 69)')
+        cy.get("label[for='gender-radio-1']").should('have.css', 'border-color', 'rgb(220, 53, 69)')
+        cy.get("#userNumber").should('have.css', 'border-color', 'rgb(220, 53, 69)')
+    })
+
+    it('cannot submit with firstName and LastName have specChar ', () => {
+
+        cy.fixture('testData').then((data) => {
+            cy.get("#firstName").type(data.firstName + '@#')
+            cy.get("#lastName").type(data.lastName + '$%^')
+            cy.get("label[for='gender-radio-1']").click();
+            cy.get("#userNumber").type(data.userNumber)
+
+            cy.get("#submit").click({ force: true })
+
+            cy.get("#firstName").should('have.css', 'border-color', 'rgb(220, 53, 69)')
+            cy.get("#lastName").should('have.css', 'border-color', 'rgb(220, 53, 69)')
+        })
+    })
+
+    it('cannot submit with invalid mobile ', () => {
```

Comment:
Another option:
```js
testData.invalidNumbers.forEach((mobile) => {
    it(`Cannot submit with invalid mobile value ${mobile}`, () => {
        // Implement here
    });
});
```

**_Pros_**: When a test data in list make the case fail, you can easily see that which data cause error

**_Cons_**: If you have 10 data, it will turn into 10 test cases